ZipIterator.hpp

```diff
@@ -53,6 +59,27 @@ class ZipRef {
   #undef OPERATOR
 };
 
+namespace std {
+
+template<std::size_t N, typename...T>
+struct tuple_element<N, ZipRef<T...>> {
+    using type = decltype(std::get<N>(std::declval<ZipRef<T...>>().val()));
+};
+
+template<typename...T>
+struct tuple_size<ZipRef<T...>>: public std::integral_constant<std::size_t, sizeof...(T)> {};
+
+template<std::size_t N, typename...T>
+decltype(auto) get(ZipRef<T...> &r) {
+    return r.template get<N>();
+}
+template<std::size_t N, typename...T>
+decltype(auto) get(const ZipRef<T...> &r) {
+    return r.template get<N>();
+}
+
+} // namespace std
+
```

Comment:
I am not a huge fan of opening `namespace std`, but you are only adding templates and each of those comes `ZipRef` type in it, so I think it's fine.